### PR TITLE
Enable removing images by digest

### PR DESCRIPTION
**Pull request by kaisoz**
Change-type: minor
Signed-off-by: Tomás Tormo <tomast@balena.io>

### CODE REVIEW

File: lib/index.coffee [hunk 1]
```diff
@@ -76,15 +76,15 @@ class DockerGC
...
-		if image.repoTags? and image.repoTags.length > 0
-			# Docker will complain if we delete by id an image referenced by more
-			# than one repository
-			Promise.each image.repoTags, (tag) =>
-				console.log("GC: Removing image: #{tag} (#{image.id})")
-				@docker.getImage(tag).remove(noprune: true)
-		else
-			console.log("GC: Removing image: #{image.id}")
-			@docker.getImage(image.id).remove(noprune: true)
+		return this.tryRemoveImageBy(image, image.repoTags) 	   ||
```
Comment (Page-) [inline lib/index.coffee:79]: Can you remove the excess whitespace before || - I assume it lines up in your editor with your settings but it definitely doesn't in a lot of other envs

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "lint": "resin-lint lib test tools",
     "pretest": "npm run lint",
     "test": "mocha --compilers coffee:coffee-script/register test",
+    "test:local": "env LOCAL_TESTS=true npm test",
```
Comment (Page-) [inline package.json:9]: Why the `env`? I've always seen it written as
```suggestion
    "test:local": "LOCAL_TESTS=true npm test",
```

File: lib/index.coffee (hunk 1 above)
Comment (kaisoz) [reply lib/index.coffee:79]: :+1:

File: package.json (hunk 2 above)
Comment (kaisoz) [reply package.json:9]: I've read that is a best practice to ensure portability... it works both ways though

File: test/index.coffee [hunk 3]
```diff
@@ -55,30 +133,33 @@ describe 'Garbage collection', ->
...
 		this.timeout(600000)
 		return Promise.resolve() if SKIP_GC_TEST
 
 		docker = @docker
 
-		# Get the size of the first image, so we can add one to it to remove
-		# the next one in addition
 		Promise.each IMAGES, (image) ->
 			pullAsync(docker, image)
+# First inspect the images we expect to be removed to make more likely that event times are recorded
```
Comment (Page-) [inline test/index.coffee:147]: Can you fix the indentation of this please

File: test/index.coffee [hunk 4]
```diff
@@ -55,30 +133,33 @@ describe 'Garbage collection', ->
...
 			docker.getImage(IMAGES[0]).inspect().get('Size')
 		.then (size) =>
 			@dockerStorage.garbageCollect(size + 1)
 		.then ->
 			Promise.map IMAGES, (image) ->
 				promiseToBool(docker.getImage(image).inspect())
-		.then (imgs) ->
-			if not _.isEqual(imgs, [false, false, true])
-				throw new Error('Incorrect images removed')
+		.then (images_found) ->
```
Comment (Page-) [inline test/index.coffee:161]:
```suggestion
		.then (imagesFound) ->
```

File: test/index.coffee (hunk 3 above)
Comment (kaisoz) [reply test/index.coffee:147]: Sure, it didn't appear in the diff :?

File: test/index.coffee (hunk 4 above)
Comment (kaisoz) [reply test/index.coffee:161]: :man_facepalming:

**Pull request merged**